classes/core/PKPContainer.php

```diff
@@ -344,6 +342,16 @@ public function runningUnitTests()
     {
         return false;
     }
+
+    /**
+     * Determine if the application is currently down for maintenance.
+     *
+     * @return bool
+     */
+    public function isDownForMaintenance()
```

Comment:
Is there a reason for the `isDownForMaintenance` / `isUnderMaintenace` discrepancy?

Reply:
right now there is none . This was done according to the implementation of the `Illuminate\Queue\QueueServiceProvider::registerWorker()` such as 

```php
$this->app->singleton('queue.worker', function ($app) {
    $isDownForMaintenance = function () {
        return $this->app->isDownForMaintenance();
    };

    $resetScope = function () use ($app) {
        if (method_exists($app['log']->driver(), 'withoutContext')) {
            $app['log']->withoutContext();
        }

        if (method_exists($app['db'], 'getConnections')) {
            foreach ($app['db']->getConnections() as $connection) {
                $connection->resetTotalQueryDuration();
                $connection->allowQueryDurationHandlersToRunAgain();
            }
        }

        $app->forgetScopedInstances();

        return Facade::clearResolvedInstances();
    };

    return new Worker(
        $app['queue'],
        $app['events'],
        $app[ExceptionHandler::class],
        $isDownForMaintenance,
        $resetScope
    );
});
```

basically a lot of laravel's core service providers depend on a container method `isDownForMaintenance()` which we will be need to override as it's not defined out container . This can be removed right now as I am overriding that method as it also require a logger and we dont have one but eventually we will need that method to be included in our container if we want to avoid overriding for more services that we may include in future .